sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/joins.scala

```diff
@@ -127,10 +128,19 @@ object ReorderJoin extends Rule[LogicalPlan] with PredicateHelper {
  * - full outer -> right outer if only the right side has such predicates
  *
  * 2. Removes outer join if it only has distinct on streamed side
+ *
  * {{{
  *   SELECT DISTINCT f1 FROM t1 LEFT JOIN t2 ON t1.id = t2.id  ==>  SELECT DISTINCT f1 FROM t1
  * }}}
  *
+ * 3. Remove outer join if aggregate references are coming from streamed side and
```

Comment:
Can we combine the doc of points 2 and 3? Let's define "Duplicate Agnostic" in the classdoc.